internal/app/secretless/handlers/mysql/handler.go

```diff
@@ -11,48 +11,127 @@ import (
 	plugin_v1 "github.com/cyberark/secretless-broker/pkg/secretless/plugin/v1"
 )
 
-// BackendConfig stores the connection info to the real backend database.
+// ConnectionDetails stores the connection info to the real backend database.
 // These values are pulled from the handler credentials config
-type BackendConfig struct {
+type ConnectionDetails struct {
 	Host     string
 	Port     uint
 	Username string
 	Password string
 	Options  map[string]string
 }
 
-func (bc *BackendConfig) Address() string {
-	return bc.Host + ":" + strconv.FormatUint(uint64(bc.Port), 10)
+// TODO: all these comments will be moved and subsumed by kumbi's work
+// This format of this packet is described here:
+//
+//   https://dev.mysql.com/doc/internals/en/mysql-packet.html
+//
+//  +-------------+----------------+---------------------------------------------+
+//  |    Type     |      Name      |                 Description                 |
+//	+-------------+----------------+---------------------------------------------+
+//  | int<3>      | payload_length | Length of the payload. The number of bytes  |
+//  |             |                | in the packet beyond the initial 4 bytes    |
+//  |             |                | that make up the packet header.             |
+//  | int<1>      | sequence_id    | Sequence ID                                 |
+//  | string<var> | payload        | [len=payload_length] payload of the packet  |
+//  +-------------+----------------+---------------------------------------------+
+//
+type mysqlPacket []byte
+
+func (pkt *mysqlPacket) SequenceId() byte {
+	return (*pkt)[3]
+}
+
+func (pkt *mysqlPacket) SetSequenceId(id byte) {
+	(*pkt)[3] = id
+}
+
+type MySQLConnection struct {
+	conn net.Conn
+	sequenceId byte
+}
+
+
+func NewMySQLClientConnection(conn net.Conn) *MySQLConnection {
```

Comment:
why not just have one method `NewConnection` that accepts a sequenceId parameter?

Reply:
though there is something about this that's off - why is the sequenceId so intricately tied to the connection?

Reply:
> why not just have one method `NewConnection` that accepts a sequenceId parameter?

You definitely do not want to do that.   That would make your client code (in the sense of "code that uses this code") aware of protocol impl details.  To the client code there is only a "client" connection and a "backend" connection.  That's how it thinks of it.  

This information may *really* belong in the protocol code I'm working  on -- you're right to notice it's not quite right that *this* protocol detail lives here and all the rest will be tidily hidden away.  I'll have to think about that.  What you see now is a cleanup of the previous code, where this mixing was much worse, so that old design may still be exerting some pressure on my thinking.  Worse case, this will documented much better with comments.  Best case, I'll find a way to bring all the protocol code together neatly.

Reply:
there's something interesting here about the connection between the `packet` (which has a sequenceId) and the `connection` (which right now also has a sequenceId). I feel this property belongs more with the packet, but it will be interesting if we can find a way to define things so that the implementation remains correct while the objects are more logical

Reply:
> I feel this property belongs more with the packet

@izgeri I see why'd you'd think so, because the data does live in the packet.  And there _is_ more cleanup to be done here probably (see below).  But I do believe it's a higher level concern than the packet.

The MySQL protocol is a stateful protocol, and as the mysql server and client send packets back and forth to each other they track what part of the process they're in using this sequence id.  That seq id is part of the _header_ (which we don't _yet_ have a proper abstraction for -- part of the other refactor).  But in any case, we want this part of the protocol to be transparent to the code that is writing the packet _payloads_ -- ie, the ConnectionPhase stuff.  I just want to say "write" this payload, and have the header with the proper sequence id inserted.  Which is why it makes sense to live at the connection level.  

Or more simply, the sequence id is, conceptually, a property of _the entire process of sending packets back and forth_ -- not of any particular packet (even though each packet does include that information).  The "Connection" is precisely what represents that entire process of sending back and forth.